cmd/chartsvc/handler.go

```diff
@@ -200,6 +200,28 @@ func getChartVersionValues(w http.ResponseWriter, req *http.Request, params Para
 	w.Write([]byte(files.Values))
 }
 
+// listChartsWithFilters returns the list of repos that contains the given chart and the latest version found
+func listChartsWithFilters(w http.ResponseWriter, req *http.Request, params Params) {
+	db, closer := dbSession.DB()
+	defer closer()
+
+	var charts []*models.Chart
+	if err := db.C(chartCollection).Find(bson.M{
+		"name": params["chartName"],
+		"chartversions": bson.M{
+			"$elemMatch": bson.M{"version": req.FormValue("version"), "appversion": req.FormValue("appversion")},
+		}}).Select(bson.M{
+		"name": 1, "repo": 1,
+		"chartversions": bson.M{"$slice": 1},
+	}).All(&charts); err != nil {
+		log.WithError(err).Errorf("could not find the given chart with id %s", params["chartName"])
```

Comment:
I think this error message was copied and pasted, it should probably just say something generic like "could not find charts with filters"